File: butler_offline/core/time.py
import datetime
from butler_offline.core import time
from typing import Callable
from datetime import date
from datetime import datetime


TODAY: Callable[[], date] = lambda: datetime.now().date()
NOW: Callable[[], datetime] = lambda: datetime.now()


def today() -> date:
    return TODAY()


def now() -> datetime:
    return NOW()


def stub_today_with(new_today):
    time.NOW = lambda: datetime(new_today.year, new_today.month, new_today.day)
    time.TODAY = lambda: new_today


def reset_viewcore_stubs():
    time.TODAY = lambda: datetime.now().date()
    time.NOW = lambda: datetime.now()

File: butler_offline/src/io/time.rs
use crate::model::primitives::datum::Datum;

#[cfg(not(feature = "integration_test"))]
use chrono::{Datelike, Local};

#[cfg(feature = "integration_test")]
use chrono::Local;

#[cfg(not(feature = "integration_test"))]
pub fn today() -> Datum {
    let time = Local::now();
    Datum::new(time.day(), time.month(), time.year())
}

#[cfg(feature = "integration_test")]
pub fn today() -> Datum {
    Datum::new(22, 1, 2019)
}

pub fn now() -> String {
    let time = Local::now();
    format!("{}", time.timestamp())
}
